src/routers/alpha-router/alpha-router.ts

```diff
@@ -562,11 +562,11 @@ export class AlphaRouter
     }
 
     let gasPriceProviderInstance: IGasPriceProvider;
-    if (this.provider instanceof JsonRpcProvider) {
+    if (JsonRpcProvider.isProvider(this.provider)) {
       gasPriceProviderInstance = new OnChainGasPriceProvider(
         chainId,
-        new EIP1559GasPriceProvider(this.provider),
-        new LegacyGasPriceProvider(this.provider)
+        new EIP1559GasPriceProvider(this.provider as JsonRpcProvider),
```

Comment:
is casting necessary here? not sure if this could cause issues / if there's a different way to avoid this

Reply:
it was necessary to satisfy the type checker